index.js

```diff
@@ -1,39 +1,342 @@
-const readline = require('readline');
-const readlineInterface = readline.createInterface(process.stdin, process.stdout);
+/**
+ *  Week 2 BCA Project:
+ *    Zorkington - a text based adventure game played
+ *    using the console modeled after the game 'Zork'
+ *    Author: Nick Castle, Connor Greenbaum
+ *    Date: 06/13/19
+ */
+
+// readline boilerplate code
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     readlineInterface.question(questionText, resolve);
   });
 }
+// end readline
 
-// remember the StateMachine lecture
-// https://bootcamp.burlingtoncodeacademy.com/lessons/cs/state-machines
-let states = {
-  'roomOne': { canChangeTo: [ 'roomTwo' ] },
-  'roomTwo': { canChangeTo: [ 'roomThree' ] },
-  'roomThree': { canChangeTo: [ 'roomOne' ] }
-};
+/**** Classes  ****/
 
-let currentState = "green";
+// class to hold game state data
+class GameState {
+  constructor(userName) {
+    this.currentRoom = rooms.Main;
+    // current player
+    this.user = new Player(userName);
+  }
+
+  moveToRoom(nextRoom) {
+    this.currentRoom = rooms[nextRoom];
+  }
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
+  printState() {
+    console.log(this);
+  }
+}
+
+// Room class
+class Room {
+  constructor(roomName, inventory) {
+    this.roomName = roomName;
+    this.inventory = inventory || [];
+    this.message = roomMessages[this.roomName];
```

Comment:
Why aren't you passing a message into the constructor? If you created the message at the same time as the rest of the object, you could eliminate the `roomMessages` object entirely.